### PR TITLE
CLUSTERMON-2155: Added basic support for infra-agent and cluster-collector to consume configs through file.

### PR DESCRIPTION
**Pull request by Sprakhar97**
Operator will create a config file to be consumed by infra agent in the infra agent directory and
another config file for cluster-collector in the collector directory derived from cluster-collector spec definition.

### CODE REVIEW

File: pkg/controller/clustercollector/clustercollector_controller.go [hunk 1]
```diff
@@ -31,6 +31,10 @@ const (
 	OLD_SPEC string = "cluster-collector-spec"
 	ClUSTER_MON_CONFIG_NAME string = "cluster-collector-config"
 	INFRA_AGENT_CONFIG_NAME string = "infra-agent-config"
+	INFRA_AGENT_NAME string = "Infra Structure Agent"
```
Comment (venkatteki) [inline pkg/controller/clustercollector/clustercollector_controller.go:34]: Lets keep "Infrastructure" as one word

File: pkg/controller/clustercollector/clustercollector_controller.go [hunk 2]
```diff
@@ -31,6 +31,10 @@ const (
...
+	CLUSTER_COLLECTOR string = "Cluster Monitor"
+	TYPE_COLLECTOR string = "Collector"
+	CLUSTER_COLLECTOR_PATH string = "./collectors/cluster-collector-linux-amd64"
```
Comment (venkatteki) [inline pkg/controller/clustercollector/clustercollector_controller.go:37]: will there be multiple flavors of this binary? And, do we always refer to linux-amd64?

File: pkg/apis/appdynamics/v1alpha1/clustercollector_types.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ type ClustercollectorSpec struct {
 	LogLevel           string                      `json:"logLevel,omitempty"`
 	ClusterMonEnabled  bool                        `json:"clusterMonEnabled,omitempty"`
 	SystemConfigs      InfraAgentConfig            `json:"systemConfig,omitempty"`
+	ExporterAddress    string                      `json:"kubeExporterAddress,omitempty"`
```
Comment (venkatteki) [inline pkg/apis/appdynamics/v1alpha1/clustercollector_types.go:22]: One general comment. Lets not expose the jira ticket details here

File: pkg/controller/clustercollector/clustercollector_controller.go (hunk 1 above)
Comment (Sprakhar97) [reply pkg/controller/clustercollector/clustercollector_controller.go:34]: This is the value the ia agent will be expecting in the agent config file for the name of the agent.

File: pkg/controller/clustercollector/clustercollector_controller.go (hunk 2 above)
Comment (Sprakhar97) [reply pkg/controller/clustercollector/clustercollector_controller.go:37]: It depends on the cluster collector image but we should make it consistent and change it to cluster-collector for all binaries, currently the image I build had the cluster collector binary with that name

**Review by harekal7 — APPROVED** [review]

**Pull request merged**